File: lcof/119/set.swift
class Solution {
    
    func longestConsecutive(_ nums: [Int]) -> Int {
        var set = Set(nums), opt = 0
        while let n = set.popFirst() {
            var prev = 0, next = 0
            while set.remove(n-1 - prev) != nil {
                prev += 1
            }
            while set.remove(n+1 + next) != nil {
                next += 1
            }
            opt = max(opt, 1 + prev + next)
        }
        return opt
    }
}

File: Algorithms/[1k, 2k)/1044/set.go
func longestDupSubstring(s string) string {
	opt := ""
	for l, r := 1, len(s)+1; l < r; {
		set, m := map[string]struct{}{}, (l+r)/2
		for i := 0; i+m <= len(s); i++ {
			str := s[i : i+m]
			_, ok := set[str]
			if ok {
				opt = str
				break
			}

			set[str] = struct{}{}
		}

		if len(opt) >= m {
			l = m + 1
		} else {
			r = m
		}
	}
	return opt
}